numpy/lib/tests/test__iotools.py

```diff
@@ -164,30 +162,36 @@ def test_upgrade(self):
         status_offset = int(nx.dtype(nx.integer).itemsize < nx.dtype(nx.int64).itemsize)
```

Comment:
This should be `nx.int_`, not `nx.integer`, and the comment should say `long`, I think.

Reply:
Curiously, `numpy.integer` is valid and is a long. Probably an obsolete usage going back to Numeric.

Reply:
`np.integer` is the base class of `np.int_` and friends. But as you observe, `np.dtype(np.integer)` promotes to `long` in the same way that `np.dtype(np.number)` promotes to `float`